src/DnsUtils.ts

```diff
@@ -0,0 +1,53 @@
+import DnsRecordsError, { DnsRecordsErrorCode } from "./errors/dnsRecordsError";
+import { CryptoRecords, DnsRecord, DnsRecordType } from "./publicTypes";
+import { isStringArray } from "./utils";
+
+export default class DnsUtils {
+
+  public toClassical(record: CryptoRecords): DnsRecord[] {
+    const dnsTypes = this.getAllDnsTypes(record);
+    return ([] as DnsRecord[]).concat(
+      ...dnsTypes.map(type => this.constructDnsRecords(record, type))
+    );
+  }
+
+  public toCrypto(records: DnsRecord[]): CryptoRecords {
+    const cryptoRecords:CryptoRecords = {};
+    for (const record of records) {
+      const { ttl, type, value } = record;
+      
+      const dnsRecord = cryptoRecords[`dns.${type}`];
+      const ttlInRecord = cryptoRecords[`dns.${type}.ttl`] as number | undefined;
+
+      if (isStringArray(dnsRecord)) {
+        dnsRecord.push(value);
+      } else {
+        cryptoRecords[`dns.${type}`] = [value];
+      }
+
+      if (!!ttlInRecord && ttlInRecord !== ttl) {
+        throw new DnsRecordsError(DnsRecordsErrorCode.NotCommonTtl, {recordType: type});
```

Comment:
Shall throw an error or pick the lowest TTL in the RRSet, according to the rfc2181 spec (the case when a client receives a malformed response from an authoritative server if we treat the smart contract as an authoritative source)?
https://tools.ietf.org/html/rfc2181#section-5.2

Reply:
I think it is fine to throw.

Reply:
I'm not entirely sure why our library needs this, and what defines its behaviour requirements. For example, what's the use-case of this method and where does this JSON schema come from?
```
    [
      { ttl: 90, value: '10.0.0.1', type: 'A' },
      { ttl: 90, value: '10.0.0.2', type: 'A' },
      { ttl: 128, value: '10.0.0.120', type: 'AAAA' }
    ]
```
My concern is that we're probably introducing something that won't be used the way it works now. But we're adding this to our public interface, which carries a promise to not break things later

Reply:
I don't think there can be huge problems in the future. It is just a helper function that makes work with DNS records from smart contracts easier. I can make it internal instead of the public if you think that no one is going to use these functions.

Reply:
Who is the user of this function then? I don't think it's us, because it's never used in our codebase outside of the tests. I assume we made it public for our library users, but I just can't see anyone who'll use it. This function assumes a certain JSON format, but I can't find it in any standard or documentation. Neither Cloudflare nor Google docs about DoH mention it. I wasn't able to find anything like it while looking through IETF drafts and RFCs as well.

Reply:
Or is it just a general function for converting a flat structure into a key-value one, that doesn't assume a certain JSON format? Like for UI developers